app/components/alchemy/admin/resource/table.rb

```diff
@@ -0,0 +1,148 @@
+          Action.new(name, tooltip, &block)
+        end
+
+        erb_template <<~ERB
+          <% if collection.any? %>
+            <table class="list">
+              <thead>
+                <tr>
+                  <% headers.each do |header| %>
+                    <%= header %>
+                  <% end %>
+                  <% if actions? %>
+                    <th class="tools"></th>
+                  <% end %>
+                </tr>
+              </thead>
+              <tbody>
+                <% collection.each do |resource| %>
+                  <tr class="<%= cycle('even', 'odd') %>">
+                    <% cells.each do |cell| %>
+                       <%= render cell.with_resource(resource) %>
+                    <% end %>
+                    <% if actions? %>
+                      <td class="tools">
+                        <% actions.each do |action| %>
+                          <%= render action.with_resource(resource) %>
+                        <% end %>
+                      </td>
+                    <% end %>
+                  </tr>
+                <% end %>
+              </tbody>
+            </table>
+          <% else %>
+            <alchemy-message type="info">
+              <%= nothing_found_label %>
+            </alchemy-message>
+          <% end %>
+        ERB
+
+        def initialize(collection, query: nil, nothing_found_label: Alchemy.t("Nothing found"), search_filter_params: {})
+          @collection = collection
+          @query = query
+          @nothing_found_label = nothing_found_label
+          @search_filter_params = search_filter_params
+        end
+
+        def column(name, label: nil, sortable: false, type: nil, alignment: nil, &block)
+          label ||= resource_handler.model.human_attribute_name(name)
+          type ||= resource_handler.model.columns_hash[name.to_s]&.type
+          attribute = resource_handler.attributes.find { |item| item[:name] == name.to_s } || {name: name, type: type}
+          block ||= lambda { |item| render_attribute(item, attribute) }
+
+          css_classes = [name, type, alignment].compact.join(" ")
+          with_header(name, @query, css_classes: css_classes, label: label, type: type, sortable: sortable)
+          with_cell(css_classes, &block)
+        end
+
+        def icon_column(icon = nil, style: nil)
+          column(:icon, label: "") do |resource|
+            render_icon(icon || yield(resource), size: "xl", style: style)
+          end
+        end
+
+        def delete_button(tooltip: Alchemy.t("Delete"), message: Alchemy.t("Are you sure?"))
+          with_action(:destroy, tooltip) do |row|
+            helpers.delete_button(resource_path(row, search_filter_params), {message: message})
+          end
+        end
+
+        def edit_button(tooltip: Alchemy.t("Edit"), size: resource_window_size)
+          with_action(:edit, tooltip) do |row|
+            link_to_dialog render_icon(:edit),
```

Comment:
Some helpers are not delegated, but this is. Why is this delegated and not taken from `helpers`?

Reply:
I changed that